planner/plansdk/analyzer.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package plansdk
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/bmatcuk/doublestar/v4"
+	"go.jetpack.io/devbox/cuecfg"
+)
+
+// Analyzers help understand the source code present in a given directory
+// Handy when implementing new Planners that need to analyze files in order
+// to determine what to do.
+type Analyzer struct {
+	rootDir string
+}
+
+func NewAnalyzer(rootDir string) *Analyzer {
+	abs, err := filepath.Abs(rootDir)
+	if err != nil {
+		abs = rootDir
```

Comment:
🤔 why not error in this case? if a rootDir cannot be converted to an absolute-dir here, then we may run into issues later on.